src/System.Net.Http/src/System/Net/Http/StreamContent.cs

```diff
@@ -12,19 +11,30 @@ namespace System.Net.Http
 {
     public class StreamContent : HttpContent
     {
-        private const int defaultBufferSize = 4096;
+        private const int DefaultBufferSize = 4096;
 
         private Stream _content;
+        private CancellationToken _cancellationToken;
         private int _bufferSize;
         private bool _contentConsumed;
         private long _start;
 
         public StreamContent(Stream content)
-            : this(content, defaultBufferSize)
+            : this(content, DefaultBufferSize)
         {
         }
 
         public StreamContent(Stream content, int bufferSize)
+            : this(content, bufferSize, CancellationToken.None)
+        {
+        }
+
+        internal StreamContent(Stream content, CancellationToken cancellationToken)
```

Comment:
>new StreamContent(decompressedStream, state.CancellationToken);

It's rather un-intuitive to pass in a cancellation token to a constructor like StreamContent. What operation needs to be cancelled? So, I do not think this ctor overload should be public at all. I have some other ideas I'm thinking about, to flow the token to the real operation which is the StreamToStream copy operation.